test/core/end2end/tests/max_message_length.cc

```diff
@@ -596,7 +596,7 @@ static void test_max_receive_message_length_on_compressed_request(
                                 nullptr);
   GPR_ASSERT(GRPC_CALL_OK == error);
 
-  cqv.Expect(tag(102), false);
+  cqv.Expect(tag(102), true);
```

Comment:
Are we sure this behavior is correct?  If the message is too large, it seems like the batch should fail.

Reply:
Documentation for `GRPC_OP_RECV_CLOSE_ON_SERVER`:

```
This operation always succeeds, meaning ops paired with
this operation will also appear to succeed, even though they may not
have.
```

Reply:
it was not, but woah nelly the only test we have covering some substantial api surface (fixed, please see pipe.h especially, but also changes in call.cc and promise_based_filter.cc)

Reply:
Ouch!  Any thoughts on how we can increase test coverage here?

For now, maybe add a comment here explaining that this is currently the only test that covers this, and that it's very important that this expected value not change, since that would break a lot of things?

I think I follow the gist of the changes to fix this.  Thanks for digging into this!